dask/utils.py

```diff
@@ -388,6 +388,9 @@ def file_size(fn, compression=None):
         with open(fn, 'rb') as f:
             f.seek(-4, 2)
             result = struct.unpack('I', f.read(4))[0]
+    elif compression:
+        with open(fn, 'rb', compression) as f:
+            result = f.seek(0, 2)
```

Comment:
It is based on the semantics of seek, so if the compressed IO class wrapper implements seek correctly. And the I added a `file_size` test to make sure it works. If someone adds another compression he should find the test when greping for current ones.